### ISSUE TITLE
Get rid of parameter 't'

### ISSUE THREAD

**Issue by oliver-batchelor** [ezyang/reflex-backpack#1]
Probably I am missing some obvious details. But in my mind backpack would allow me to totally get rid of the whole 't' parameter, with plain data types, e.g.

Event :: Type -> Type
Behavior :: Type -> Type

hold :: MonadHold m => a -> Event a -> m (Behavior a)

**Comment by ezyang** [comment]
Hello @Saulzar,

When I originally Backpack'ed reflex, I indeed remove the 't' parameter. But @ryantrinkle pointed out to me that the t parameter plays an important role for the Spider, because they support the rank-2 types that prevents people from mixing up different instances of Spider.

But maybe this is all moot and we should just use the global timeline in all cases. Anyway, it's a good idea for another "Backpack version" of Reflex.

**Comment by ryantrinkle** [comment]
Basically, the `t` parameter plays a dual role: selecting the implementation and distinguishing independent data flow graphs.  The first role is not necessary with groundhog (unless you want to be able to be truly polymorphic in implementation).  However, second still is.  In particular, Spider is single-threaded, so if you want to have two Spider-based graphs executing in parallel, you need to ensure that they're unconnected.  Currently, the only way to do this is to ensure that their `t` parameters are different.  It functions much like the parameter to the `ST` monad.

**Comment by oliver-batchelor** [comment]
Yeah, I see - that makes a lot of sense. Still, I can imagine the tradeoff
for simplicity vs safety might be worth it...

Unlike ST where it's highly likely you might use more than one instance,
most client side apps would almost never have two Spider instances.

**Comment by ryantrinkle** [comment]
@Saulzar You're quite right.  To me, the ideal would be to have some way for people who don't know/care about the distinction to have it hidden, while keeping their code compatible with more "advanced" usage that the parameter allows.  I'm not exactly sure how to achieve that, though!
